src/DAL/Orders/Strategies/OptimisticConcurrencyBookingStrategy.cs

```diff
@@ -0,0 +1,42 @@
+﻿using DAL.Events;
+using DAL.Orders.Repository;
+
+namespace DAL.Orders.Strategies
+{
+	internal class OptimisticConcurrencyBookingStrategy : IExecutionStrategy<Guid>
+	{
+		private readonly ICartRepository _cartRepository;
+
+		public OptimisticConcurrencyBookingStrategy(ICartRepository cartRepository)
+		{
+			_cartRepository = cartRepository;
+		}
+
+		public async Task<bool> Execute(Guid id)
+		{
+			try
+			{
+				await _cartRepository.BeginTransaction();
+
+				var seats = (await _cartRepository.GetItemsWithEventSeat(id)).Select(ci => ci.EventSeat).ToList();
+				if (seats.Any(s => !string.Equals(s.Status, SeatStatus.Available.ToString(),
```

Comment:
If you stored seatState as an integer then you would not need to convert each value to a string. Also with number value in DB, it is easier to manage state names with time to come. If you decide for example to rename one of your states - you will need to prepare a script to update all the history. In the case of numbers in DB - you just need to rename your Enum.